mxcube3/routes/Beamline.py

```diff
@@ -254,3 +254,17 @@ def beamline_get_data_path():
     """
     data = mxcube.session.get_base_image_directory()
     return jsonify(data)
+
+@mxcube.route("/mxcube/api/v0.1/beamline/prepare_beamline", methods=['PUT'])
```

Comment:
What does prepare beamline do, could it be generic to all beamines ?

We maybe don't need to send this explicitly from the client but we can deduce when its needed in the server, like
we call it when a sample is added, like this we don't need any client side implementation of this at all ?

Reply:
In our case, in our Collect hwobj
- Move detector back
- Close safety shutter
- Diff to transfer

We trigger from the client in order to do those steps before the sample is actually added to the queue. We saw (in the previous user run, but not commited this before :( ) that the users were more aware of that this way.

Reply:
Hm, this should in my opinion definitely be handled by the control or hardware object layer. Maybe even in the collect hwobj or in the sample changer hw object.

Adding such logic in the client makes the behavior of the beamline control dependent on the UI.

Reply:
We can discuss about this, but for me the beamline acts on a user request, so I do not see an issue. Is later on the hwobj level where it is triggered.

Reply:
In a way everything acts on a users request, this needs to be pushed back to the backend or better to the hardware object layer, of there are no really good reasons for not doing that.